### PR TITLE
CI: switch to Travis-CI container mode

### PR DESCRIPTION
**Pull request by pv**
Try switching to Travis new container-based infra. Disables sudo, but enables ccache, which might speed up our testing times.

### CODE REVIEW

**Comment by pv** [comment]
Looks promising --- 'fast' test suite takes 10 min instead of the previous 20-25min when everything is in ccache.

**Comment by pv** [comment]
Going to merge, about twice faster.

File: .travis.yml [hunk 1]
```diff
@@ -46,25 +47,52 @@ matrix:
...
+    - gfortran
+    - libgmp-dev
+    - libmpfr-dev
+    - ccache
+cache:
+  directories:
+    - $HOME/.ccache
 before_install:
-  # Automated travis-ci wheel builds (disabled because intermittent issues)
+  - export PATH=/usr/lib/ccache:$PATH
```
Comment (larsmans) [inline .travis.yml]: Why is this?

**Comment by larsmans** [comment]
The speed improvement is impressive!

File: .travis.yml (hunk 1 above)
Comment (pv) [inline .travis.yml]: The `ccache` binaries that override `gcc` etc. are there on Debian.

**Pull request merged**